src/components/Form.js

```diff
@@ -219,6 +219,10 @@ export default class Form extends Component {
   };
 
   onSubmit = event => {
+    if (event.target !== event.currentTarget) {
```

Comment:
Should we move `event.preventDefault()` to the beginning of this function? (this way, if the first condition is passed and `return;` is called, the default event handler for the form is still not called).